dags/analyzer_helper/tests/unit/test_unit_discord_transform_raw_data.py

```diff
@@ -0,0 +1,129 @@
+import unittest
+from datetime import datetime
+
+from analyzer_helper.discord.discord_transform_raw_data import DiscordTransformRawData
+
+
+class DiscordTransformRawDataUnitTest(unittest.TestCase):
+
+    def test_create_interaction_valid_data(self):
+        transformer = DiscordTransformRawData()
+        interaction = transformer.create_interaction("reply", ["user1234"], "emitter")
+        self.assertEqual(
+            interaction,
+            {
+                "name": "reply",
+                "users_engaged_id": ["user1234"],
+                "type": "emitter",
+            },
+        )
+
+    def test_create_interaction_missing_arguments(self):
+        transformer = DiscordTransformRawData()
+        with self.assertRaises(TypeError):
+            transformer.create_interaction("reply")
+
+    def test_create_receiver_interaction_valid_data(self):
+        transformer = DiscordTransformRawData()
+        data = {
+            "createdDate": {"$date": "2024-06-11T00:00:00Z"},
+            "messageId": "12345",
+            "threadId": "abc",
+            "channelId": "xyz",
+            "isGeneratedByWebhook": False,
+            "botActivity": False,
+        }
+        interaction = transformer.create_receiver_interaction(
+            data, "reply", "user456", "user789"
```

Comment:
Please add the equal operator for function parameters. It would increase the code understability.